### PR TITLE
feat(web-server): add enclave attestation report endpoint

### PR DESCRIPTION
**Pull request by PiDelport**
- Issue #1

PRs:

- Follows #49
- Precedes #68

### CODE REVIEW

File: web-server/sgx-wallet/enclave/Makefile [hunk 1]
```diff
@@ -39,7 +39,7 @@ FILES_T_F_RUST=$(addprefix $(SRC_T), $(FILES_T_RUST))
 OUTPUT_W_FT=$(addprefix $(OBJ_T), $(OUTPUT_T))
 
 # All Rust and other source files that the Cargo build depends on.
-FILES_RUST_F = Cargo.toml Cargo.lock $(shell find src -name '*.rs')
+FILES_RUST_F = Cargo.toml Cargo.lock $(shell find src ../../sgx-wallet-impl/src -name '*.rs')
```
Comment (Nghondzweni) [inline web-server/sgx-wallet/enclave/Makefile:42]: Was this finding `src` from `web-client`?

Comment (PiDelport) [reply web-server/sgx-wallet/enclave/Makefile:42]: Nah, these two roots refer to:

* `web-server/sgx-wallet/enclave/src`
* `web-server/sgx-wallet-impl/src`

File: web-server/http-service-impl/src/server.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+pub async fn run_server(
+    wallet_enclave: Box<dyn WalletEnclave>,
+    bind_addr: &str,
+) -> std::io::Result<()> {
+    let enclave_arbiter = Arbiter::new();
+    let wallet_enclave_addr = Actor::start_in_arbiter(&enclave_arbiter.handle(), |_ctx| {
+        WalletEnclaveActor { wallet_enclave }
+    });
+
+    let server = HttpServer::new(move || {
```
Comment (Nghondzweni) [inline web-server/http-service-impl/src/server.rs:23]: Does `move` change the port if it's taken?

**Review by Nghondzweni — APPROVED** [review]

File: web-server/http-service-impl/src/server.rs (hunk 2 above)
Comment (PiDelport) [reply web-server/http-service-impl/src/server.rs:23]: Nah, this is just for data ownership: the closure gets saved and used to spawn new handlers, so it has to own the values it refers to (particularly `wallet_enclave_addr`).

File: web-server/sgx-wallet/enclave/Makefile (hunk 1 above)
Comment (Nghondzweni) [reply web-server/sgx-wallet/enclave/Makefile:42]: Oh, got you.

**Review by longtomjr — APPROVED** [review]

**Pull request merged**